### DISCUSSION TITLE
What is the use of Stepwise Paired Data?

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/fda-model#49] [Q&A] [answered]
What relationship do we represent with of stepwise paired data? is it for inflow/outflow relationships? Is there a way we could just use paired data to accomplish the same thing without adding another class? 

It seems we discussed this at one point, but I'm drawing a blank, so I propose we discuss on here one more time for the record.

**Comment by HenryGeorgist** [comment]
we can use the same battery of tests for paired data, the difference is that we should not expect interpolated values, so for the ones that test interpolation they will be actually expecting the previous y or x value.

**Comment by Brennan1994** [comment]
Makes sense. I was more thinking is there any reason to write the integrate or multiply logic for them? Or is that code just noise because you'd never integrate or multiply a step function in FDA

**Comment by HenryGeorgist** [comment]
Honestly, I dont know why we would ever use this in a real project. I think this is unnecessary complication.

**Comment by HenryGeorgist** [comment]
You can get the same basic behavior with paired data by putting in an x with a y, and then a very similar y (+.0000001) for an x that is increased by the step

**Comment by Brennan1994** [comment]
That's kind of what I was thinking, but I've also never really used FDA for anything other than investigations. Interested in Richard's take on this issue.

**Comment by rnugent3** [comment]
I can't think of a reason to use a step-wise function. When I think of step-wise functions, I am thinking of a mapping of a discrete variable to a continuous variable. While we do discretize, all of the variables we use to get the results we need are either inherently continuous or (typically) large enough to be treated as continuous.

**Comment by Brennan1994** [comment]
alright, with that input, I propose we remove stepwise paired data from the model. Happy to make a ticket pending Richard's approval.

**Comment by rnugent3** [comment, marked as answer]
I support removing stepwise from  fda-model. 

This may be one of those decision points we were just discussing. I am thinking about the user story - I don't see any situation in which a user would want to map a discrete variable to a continuous variable in FDA - so this functionality is superfluous. 

Shall we apply this decision to the original codebase/the UI work?

**Comment by Brennan1994** [comment]
Converting this issue to a discussion.

I think we should. If we don't see a reason to implement it, we shouldn't on any of the levels.
